### ISSUE TITLE
[BUG] Scheduler and/or Autoconf not connecting to Database

### ISSUE THREAD

**Issue by 0FiRE0** [bunkerity/bunkerweb#596]
**Description**
Concise description of what you're trying to do, the expected behavior and the current bug.
On a random basis, sometimes the `scheduler` and `autoconf` work and are able to connect to the `database`, sometimes it works sometimes not, although the `database` is up and running and accessible. Also sometimes only one of them fails to connect.
If it didn't work, we need to remove all containers and recreate them again `docker compose down` then `docker compose up`, and monitor the logs to check.

**How to reproduce**
Give steps on how to reproduce the bug (e.g. : commands, yaml, configs, tests, environment, version, ...).
Version: 1.5.1
Setup: Docker Compose - Autoconf

Just run `docker compose up` and monitor the logs, if it didn't work we need to run `docker compose down` then `docker compose up` without changing anything.

Noting that this also happens on the normal setup even without `autoconf`, where only the `scheduler` fails to connect to the `database`.

```
version: "3.5"

services:
  bunkerweb:
    image: bunkerity/bunkerweb:1.5.1
    ports:
      - 80:8080
      - 443:8443
    labels:
      - "bunkerweb.INSTANCE"
    environment:
      - SERVER_NAME=
      - DATABASE_URI=mariadb+pymysql://bunkerweb:123456@bw-db:3306/db
      - AUTOCONF_MODE=yes
      - MULTISITE=yes
      - API_WHITELIST_IP=127.0.0.0/8 10.20.30.0/24
      - AUTO_LETS_ENCRYPT=no
    networks:
      bw-universe:
        aliases:
          - bunkerweb
      bw-services:
        aliases:
          - bunkerweb

  bw-autoconf:
    image: bunkerity/bunkerweb-autoconf:1.5.1
    depends_on:
      - bunkerweb
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:123456@bw-db:3306/db
      - AUTOCONF_MODE=yes
      - DOCKER_HOST=tcp://bw-docker:2375
    networks:
      bw-universe:
        aliases:
          - bw-autoconf
      bw-docker:
        aliases:
          - bw-autoconf

  bw-scheduler:
    image: bunkerity/bunkerweb-scheduler:1.5.1
    depends_on:
      - bunkerweb
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:123456@bw-db:3306/db
      - DOCKER_HOST=tcp://bw-docker:2375
      - AUTOCONF_MODE=yes
    networks:
      bw-universe:
        aliases:
          - bw-scheduler
      bw-docker:
        aliases:
          - bw-scheduler

  bw-docker:
    image: tecnativa/docker-socket-proxy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - CONTAINERS=1
      - LOG_LEVEL=warning
    networks:
      bw-docker:
        aliases:
          - bw-docker

  bw-db:
    image: mariadb:10.10
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_DATABASE=db
      - MYSQL_USER=bunkerweb
      - MYSQL_PASSWORD=123456
    volumes:
      - bw-data:/var/lib/mysql
    networks:
      bw-docker:
        aliases:
          - bw-db

volumes:
  bw-data:

networks:
  bw-universe:
    name: bw-universe
    ipam:
      driver: default
      config:
        - subnet: 10.20.30.0/24
  bw-services:
    name: bw-services
  bw-docker:
    name: bw-docker
```

**Logs**
The logs generated by BunkerWeb. **DON'T FORGET TO REMOVE PRIVATE DATA LIKE IP ADDRESSES !**

This is a sample log where both the `scheduler` and the `autoconf` are not able to connect.
Sometimes it happens to one of them as well, whereas the second one is working.
```
bunkerweb-bunkerweb-1     | 2023/08/17 07:24:51 [notice] 45#45: *5 [INIT-WORKER] BunkerWeb is ready to fool hackers ! 🚀, context: ngx.timer
bunkerweb-bw-scheduler-1  | [2023-08-17 07:24:52] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-autoconf-1   | [2023-08-17 07:24:56] - AUTOCONF - ℹ️  - BunkerWeb instances are ready 🚀
bunkerweb-bw-autoconf-1   | [2023-08-17 07:24:56] - AUTOCONF - ℹ️  - Instance #1 : bunkerweb-bunkerweb-1
bunkerweb-bw-autoconf-1   | [2023-08-17 07:24:56] - AUTOCONF - ℹ️  - Processing events ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:24:57] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:02] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-autoconf-1   | [2023-08-17 07:25:04] - DOCKER-CONTROLLER - ℹ️  - Caught Docker event, deploying new configuration ...
bunkerweb-bw-autoconf-1   | [2023-08-17 07:25:05] - DOCKER-CONTROLLER - ℹ️  - Successfully deployed new configuration 🚀
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:07] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:12] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:18] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:23] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:28] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
bunkerweb-bw-autoconf-1   | [2023-08-17 07:25:29] - DOCKER-CONTROLLER - ℹ️  - Caught Docker event, deploying new configuration ...
bunkerweb-bw-autoconf-1   | [2023-08-17 07:25:29] - DOCKER-CONTROLLER - ℹ️  - Successfully deployed new configuration 🚀
bunkerweb-bw-scheduler-1  | [2023-08-17 07:25:33] - SCHEDULER - ⚠️  - Autoconf is not loaded yet in the database, retrying in 5s ...
```

**Comment by TheophileDiot** [comment]
Hi @0FiRE0, thank you for opening this issue, do you have more logs maybe to see if the autoconf managed to write inside the database ?

**Comment by TheophileDiot** [comment]
@0FiRE0 oh alright, it seems that Docker can be annoying sometimes with the dns resolution of hostnames, I'll have a look and let you know what I can find

**Comment by fl0ppy-d1sk** [comment]
Hello @0FiRE0,

It should be fixed in the last [1.5.2 version](https://github.com/bunkerity/bunkerweb/releases/tag/v1.5.2). Feel free to open a new issue if needed.
